test/parallel/test-net-connect-local-error.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+
+var client = net.connect({
+  port: common.PORT + 1,
+  localPort: common.PORT,
+  localAddress: '127.0.0.1'
+});
+
+var onErrorCalled = false;
+client.on('error', function(err) {
```

Comment:
instead of having the `onErrorCalled`, you could wrap the callback to the error event in `common.mustCall()`.

You could then remove the `process.on('exit')` listener too